sdk/core/azure-core/inc/response.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <http/http.hpp>
+
+namespace Azure { namespace Core {
+
+  template <class T> class Response {
+    T m_value;
+    std::unique_ptr<Http::RawResponse> m_rawResponse;
+
+  public:
+    // Require a raw response to create a Response T
+    Response(const T& initialValue, std::unique_ptr<Http::RawResponse> rawResponse)
```

Comment:
```suggestion
    Response(const T& initialValue, std::unique_ptr<Http::RawResponse>&& rawResponse)
```

Reply:
Since `T` is usually a complex struct, is there any way to avoid the copy of `T` here? I mean when T is rvalue.